qiskit_experiments/quantum_volume/qv_experiment.py

```diff
@@ -0,0 +1,249 @@
+from numpy.random import Generator, default_rng
+from qiskit.providers.backend import Backend
+from qiskit.providers.options import Options
+
+try:
+    from qiskit import Aer
+
+    HAS_SIMULATION_BACKEND = True
+except ImportError:
+    HAS_SIMULATION_BACKEND = False
+
+from qiskit import QuantumCircuit
+from qiskit.circuit.library import QuantumVolume
+from qiskit import transpile, assemble
+from qiskit import execute
+from qiskit.exceptions import QiskitError
+from qiskit.providers.basebackend import BaseBackend as LegacyBackend
+from qiskit_experiments.base_experiment import BaseExperiment
+from qiskit_experiments.experiment_data import ExperimentData
+from .qv_analysis import QVAnalysis
+
+
+class QVExperiment(BaseExperiment):
+    """Quantum Volume Experiment class
+
+    Experiment Options:
+        trials (int): Optional, number of times to generate new Quantum Volume circuits and
+                    calculate their heavy output.
+    """
+
+    # Analysis class for experiment
+    __analysis_class__ = QVAnalysis
+
+    # ExperimentData class for the simulations
+    __simulation_data__ = ExperimentData
+
+    def __init__(
+        self,
+        qubits: Union[int, Iterable[int]],
+        trials: Optional[int] = 1,
+        seed: Optional[Union[int, Generator]] = None,
+        simulation_backend: Optional[Backend] = None,
+    ):
+        """Quantum Volume experiment
+        Args:
+            qubits: the number of qubits or list of
+                    physical qubits for the experiment.
+            trials: number of trials to run the quantum volume circuit.
+            seed: Seed or generator object for random number
+                  generation. If None default_rng will be used.
+            simulation_backend: the simulator backend to use to generate
+                the expected results. the simulator must have a 'save_probabilities' method.
+                if None Aer simulator will be used (in case Aer is not installed -
+                qiskit.quantum_info will be used).
+        """
+        super().__init__(qubits)
+
+        # Set configurable options
+        self.set_experiment_options(trials=trials)
+
+        # Set fixed options
+        self._previous_trials = 0
+        self._simulation_data = None
+
+        if not isinstance(seed, Generator):
+            self._rng = default_rng(seed=seed)
+        else:
+            self._rng = seed
+
+        if not simulation_backend and HAS_SIMULATION_BACKEND:
+            self._simulation_backend = Aer.get_backend("aer_simulator")
+        else:
+            self._simulation_backend = simulation_backend
+
+    @classmethod
+    def _default_experiment_options(cls):
+        return Options(trials=1)
+
+    # pylint: disable = arguments-differ
+    def run(
+        self,
+        backend: "Backend",
+        analysis: bool = True,
+        experiment_data: Optional[ExperimentData] = None,
+        simulation_data: Optional[ExperimentData] = None,
```

Comment:
I just wanted to make clear that we should compare the result of the QV circuits run on a backend with the ideal expected probabilities **before** transpilation, since the QV measure both for the HW and SW parts, incl. transpilation